distributed_loader.cc

```diff
@@ -366,12 +367,20 @@ static std::vector<sstables::shared_sstable> sstables_for_shard(const std::vecto
 }
 
 static future<> populate(distributed<database>& db, sstring datadir) {
-    return lister::scan_dir(datadir, { directory_entry_type::directory }, [&db] (fs::path datadir, directory_entry de) {
-        auto& ks_name = de.name;
-        if (is_system_keyspace(ks_name)) {
+    return do_with(std::vector<future<>>(), [&db, datadir = std::move(datadir)] (std::vector<future<>>& futures) {
+        // push futures that populate keyspaces into array of futures,
+        // so that the supplied callback will not block scan_dir() from
+        // reading the next entry in the directory.
+        return lister::scan_dir(datadir, { directory_entry_type::directory }, [&db, &futures] (fs::path datadir, directory_entry de) {
+            auto& ks_name = de.name;
+            if (is_system_keyspace(ks_name)) {
```

Comment:
Super nit: I think the code will be shorter if you do:
\[ ... \]( ... ){
if (!is_system_keyspace(de.name)) {
    futures.emplace_back(distributed_loader::populate_keyspace(db, datadir.native(), de.name));
}
};

No need to return at all.

Reply:
I inverted the condition, but the `lister::scan_dir` function requires a `std::function<future<> (something)>`. Unless I return `make_ready_future<>()` at the end, the compiler complains about incompatible types.